src/daft-sql/src/planner.rs

```diff
@@ -233,33 +262,27 @@ impl SQLPlanner {
                         format_clause: None,
                     }
                 }
-                match (op, set_quantifier) {
-                    (Union, SetQuantifier::All) => {
-                        let left = self.plan_query(&make_query(left))?;
-                        let right = self.plan_query(&make_query(right))?;
-                        return left.union(&right, true).map_err(|e| e.into());
-                    }
+
+                let left = self.new_with_context().plan_query(&make_query(left))?;
+                let right = self.new_with_context().plan_query(&make_query(right))?;
```

Comment:
same question here

Reply:
Similar thing to above, the left and right queries don't(?) share alias or table maps with each other or the outer query.